src/expr/type_node.h

```diff
@@ -429,6 +429,16 @@ class TypeNode {
    */
   bool isInterpretedFinite() const;
 
+
+  /**
+   * Is this a first-class type?
+   * First-class types are types for which we handle equalities between terms
```

Comment:
This wording is a bit confusing.  Is the following rewording correct?

First-class types have the following two properties:
1) we can reason about equalities between terms of that type; and
2) they are allowed to be used to instantiate the parameters of parametric theories (e.g. the index or element types in arrays)